### PR TITLE
oneHotEncoding support numeric type

### PR DESCRIPTION
**Pull request by Huandao0812**
### Description and related tickets, documents
oneHotEncoding able to support numeric type except Vector

Reviewers: @phvu 

### PR Progress
Make sure all checkboxes below are checked before merged
- [x] Branch is in format `prefix/description` (see [this](http://www.guyroutledge.co.uk/blog/git-branch-naming-conventions/))
- [x] Merge check has no conflicts. PR checks passed.
- [x] Code review is done.

### CODE REVIEW

File: spark/src/main/scala/io/ddf/spark/etl/TransformationHandler.scala [hunk 1]
```diff
@@ -322,10 +323,17 @@ class TransformationHandler(mDDF: DDF) extends CoreTransformationHandler(mDDF) {
 
   override def oneHotEncoding(inputColumn: String, outputColumnName: String): DDF = {
     val df = this.mDDF.getRepresentationHandler.get(classOf[DataFrame]).asInstanceOf[DataFrame]
+    val newCol = this.mDDF.getColumn(inputColumn) match {
+      case x if(x.getType == Schema.ColumnType.DOUBLE) => df.col(inputColumn)
+      case x if(x.getType != Schema.ColumnType.VECTOR && x.isNumeric)
```
Comment (nhanitvn) [inline spark/src/main/scala/io/ddf/spark/etl/TransformationHandler.scala]: Oops, what column type is this? Vector!!! So each value is a Vector?

File: spark/src/main/scala/io/ddf/spark/etl/TransformationHandler.scala [hunk 2]
```diff
@@ -322,10 +323,17 @@ class TransformationHandler(mDDF: DDF) extends CoreTransformationHandler(mDDF) {
...
+        => df.col(inputColumn).cast(DoubleType)
+      case someType => throw new DDFException(s"Column type ${someType.toString} not supported for oneHotEncoding")
+    }
+    val newDF = df.withColumn(inputColumn, newCol)
     val encoder = new OneHotEncoder()
```
Comment (nhanitvn) [inline spark/src/main/scala/io/ddf/spark/etl/TransformationHandler.scala]: +1 for more Spark API usage, @Huandao0812

File: spark/src/main/scala/io/ddf/spark/etl/TransformationHandler.scala (hunk 1 above)
Comment (Huandao0812) [inline spark/src/main/scala/io/ddf/spark/etl/TransformationHandler.scala]: no, that statement mean if columnType is not Vector and is numeric, then convert it into double

Comment (nhanitvn) [inline spark/src/main/scala/io/ddf/spark/etl/TransformationHandler.scala]: I am just curious about that VECTOR type, @Huandao0812. What does the value look like?

Comment (phvu) [inline spark/src/main/scala/io/ddf/spark/etl/TransformationHandler.scala]: It is a column of Vector. Value is a vector.

**Comment by phvu** [comment]
reviewed.

**Comment by hai-adatao** [comment]
If you are done with this PR, please merge @Huandao0812

**Pull request merged**